File: compression.py
import gzip
import zlib

inputString = "blahblahblahblahblah"
print(inputString)
inputBytes = inputString.encode()
print(f"Uncompressed length: {len(inputBytes)}")

# Compress the bytes directly to a .gz file
# with gzip.open("compressed.gz", "wb") as f:
#     f.write(inputBytes)
open("compressed.gz", "wb").write(gzip.compress(inputBytes)) # For some reason this is better

# Compress the bytes
outputBytes = zlib.compress(inputBytes)
print(f"Compressed length: {len(outputBytes)}")
# print(outputBytes.hex())

# Decompress the bytes
result = zlib.decompress(outputBytes)

# Decode the bytes to a str
print(result.decode())

File: compression.js
const fs = require("node:fs");
const zlib = require("node:zlib");

const inputString = "blahblahblahblahblah";
console.log(inputString);
const inputBytes = Buffer.from(inputString);
console.log(`Uncompressed length: ${inputBytes.length}`);

// Compress the bytes directly to a .gz file
fs.writeFileSync("compressed.gz", zlib.gzipSync(inputBytes));

// Compress the bytes
const outputBytes = zlib.deflateSync(inputBytes);
console.log(`Compressed length: ${outputBytes.length}`);
// console.log(outputBytes.toString("hex"));

// Decompress the bytes
const result = zlib.inflateSync(outputBytes);

// Decode the bytes to a string
console.log(result.toString());
